### PR TITLE
Fix null dereference when debugging untitlted files

### PR DESCRIPTION
**Pull request by rjmholt**
Fixes https://github.com/PowerShell/vscode-powershell/issues/1334.

I was looking at this trying to figure out what's going on, and I think what's happening is that the null check we do it not helpful because we don't await the execution of the method.

So I've changed it to pass the actual value of the string in at call time. I assume that will work, but let me know if not.

Also the first commit here was a style change one...

### CODE REVIEW

File: src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs [hunk 1]
```diff
@@ -207,16 +201,15 @@ protected void Stop()
...
+            if (!string.IsNullOrEmpty(_scriptToLaunch))
             {
-                if (this.editorSession.PowerShellContext.SessionState == PowerShellContextState.Ready)
+                if (_editorSession.PowerShellContext.SessionState == PowerShellContextState.Ready)
                 {
                     // Configuration is done, launch the script
-                    var nonAwaitedTask =
-                        this.LaunchScript(requestContext)
-                            .ConfigureAwait(false);
+                    var nonAwaitedTask = LaunchScript(requestContext, _scriptToLaunch)
```
Comment (rjmholt) [inline src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs]: Here's the real change

**Comment by rjmholt** [comment]
Well there's a null check around the call, but I assume the object field is changed after the check but before the evaluation.

Here's the tail of the logs from the issue:
```
2018-05-25 11:22:25 [ERROR] - Method "OnListenTaskCompleted" at line 391 of C:\projects\powershelleditorservices\src\PowerShellEditorServices.Protocol\MessageProtocol\ProtocolEndpoint.cs

    ProtocolEndpoint message loop terminated due to unhandled exception:
    
    System.AggregateException: One or more errors occurred. ---> System.ArgumentNullException: Value cannot be null.
    Parameter name: path
       at Microsoft.PowerShell.EditorServices.ScriptFile.IsUntitledPath(String path)
       at Microsoft.PowerShell.EditorServices.Protocol.Server.DebugAdapter.<HandleLaunchRequest>d__26.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol.MessageDispatcher.<DispatchMessage>d__7.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol.ProtocolEndpoint.<ListenForMessages>d__36.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Microsoft.PowerShell.EditorServices.Utility.AsyncContext.Start(Func`1 asyncMainFunc, ILogger logger)
       at System.Threading.Tasks.Task.Execute()
       --- End of inner exception stack trace ---
    ---> (Inner Exception #0) System.ArgumentNullException: Value cannot be null.
    Parameter name: path
       at Microsoft.PowerShell.EditorServices.ScriptFile.IsUntitledPath(String path)
       at Microsoft.PowerShell.EditorServices.Protocol.Server.DebugAdapter.<HandleLaunchRequest>d__26.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol.MessageDispatcher.<DispatchMessage>d__7.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol.ProtocolEndpoint.<ListenForMessages>d__36.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Microsoft.PowerShell.EditorServices.Utility.AsyncContext.Start(Func`1 asyncMainFunc, ILogger logger)
       at System.Threading.Tasks.Task.Execute()<---
    

2018-05-25 11:22:25 [ERROR] - Method "ProtocolEndpoint_UnhandledException" at line 434 of C:\projects\powershelleditorservices\src\PowerShellEditorServices.Host\EditorServicesHost.cs

    PowerShell Editor Services is terminating due to an unhandled exception, see previous logs for details.
```

**Comment by tylerl0706** [comment]
Makes sense :)

**Comment by rjmholt** [comment]
To be honest I get the impression there are more than a few race conditions hanging around the codebase and sharing object fields over tasks is responsible. So a more functional style might be our saviour here

**Comment by tylerl0706** [comment]
PURE FUNCTIONS PLEASE

**Comment by rjmholt** [comment]
`[Pure]`

**Pull request merged**